src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsOnSpecialOasis.java

```diff
@@ -7,75 +7,60 @@
 import org.openqa.selenium.Dimension;
 import org.testng.annotations.Test;
 
-/**
- * https://www.google.com/dfp/5441#delivery/LineItemDetail/lineItemId=126608052
- */
+import java.util.HashMap;
 
 public class TestAdsOnSpecialOasis extends TemplateNoFirstLoad {
   static final String PROJECT43_TEST_LINE_ITEM_ID = "271491732";
 
-  @Test(groups = "TestAdsOnSpecialPagesOasis")
-  public void testAdsOnSpecialVideoPageOasis() throws Exception {
-    AdsBaseObject ads = buildAdsObjectForPage("Special:Videos");
-    verifySlotHasCorrectAd(
-        ads,
-        AdsContent.TOP_LB,
-        "wka.life/_project43//special",
-        PROJECT43_TEST_LINE_ITEM_ID
-    );
-  }
+  static HashMap<String, String> SPECIAL_PAGE_EXPECTED_LINE_ITEMS = new HashMap<>();
+  static HashMap<String, String> SPECIAL_PAGE_EXPECTED_AD_UNITS = new HashMap<>();
 
-  @Test(groups = "TestAdsOnSpecialPagesOasis")
-  public void testAdsOnSpecialImagesPageOasis() throws Exception {
-    AdsBaseObject ads = buildAdsObjectForPage("Special:Images");
-    verifySlotHasCorrectAd(
-        ads,
-        AdsContent.TOP_LB,
-        "wka.life/_project43//special",
-        PROJECT43_TEST_LINE_ITEM_ID);
-  }
+  static HashMap<String, String> FILE_PAGE_EXPECTED_LINE_ITEMS = new HashMap<>();
+  static HashMap<String, String> FILE_PAGE_EXPECTED_AD_UNITS = new HashMap<>();
+  static final String FILE_PAGE_AD_UNIT = "wka.life/_project43//file";
 
   private AdsBaseObject buildAdsObjectForPage(String pageName) {
     String testedPage = urlBuilder.getUrlForPath("project43", pageName);
     return new AdsBaseObject(driver, testedPage, new Dimension(1292, 1000));
   }
 
-  private void verifySlotHasCorrectAd(
-      AdsBaseObject ads,
-      String slotName,
-      String adUnit,
-      String lineItem
-  ) {
-    ads.verifyGptIframe(adUnit, slotName, "gpt");
-    ads.verifyGptAdInSlot(slotName, lineItem, "");
+  private void specialPageExpectedDataProvider() {
+    SPECIAL_PAGE_EXPECTED_LINE_ITEMS.put(AdsContent.TOP_LB, "271491732");
+    SPECIAL_PAGE_EXPECTED_AD_UNITS.put(AdsContent.TOP_LB, "wka.life/_project43//special");
+  }
+
+  private void testSpecialPage(AdsBaseObject ads) {
+    ads.setPageType("special");
+    specialPageExpectedDataProvider();
```

Comment:
Can we use this hashmap in test scope only? It seems not needed as static class property